src/Generator/Generators/CSharp/CSharpSources.cs

```diff
@@ -641,10 +642,16 @@ public override void GenerateClassSpecifier(Class @class)
 
             if (@class.NeedsBase)
             {
-                bases.AddRange(
-                    from @base in @class.Bases
-                    where @base.IsClass
-                    select @base.Class.Visit(TypePrinter).Type);
+                foreach (var @base in @class.Bases.Where(b => b.IsClass))
+                {
+                    var typeMaps = new List<System.Type>();
+                    var keys = new List<string>();
+                    this.DisableTypeMap(@base.Class, typeMaps, keys);
+                    var printedBase = @base.Type.Desugar().Visit(TypePrinter);
+                    bases.Add($"{printedBase}{printedBase.NameSuffix}");
+                    for (int i = 0; i < typeMaps.Count; i++)
```

Comment:
This cannot work. TypeMaps is a list of types, not of TypeMap-s. We cannot control this through such a property. We cannot use a static one either because it would enable or disable all type maps rather than just the one in question.